bson/mgocompat/setter_getter.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mgocompat
+
+import (
+	"reflect"
+
+	"go.mongodb.org/mongo-driver/bson/bsoncodec"
+	"go.mongodb.org/mongo-driver/bson/bsonrw"
+)
+
+// The Raw type represents raw unprocessed BSON documents and elements.
+// Kind is the kind of element as defined per the BSON specification, and
+// Data is the raw unprocessed data for the respective element. This is
+// identical to mgobson's Raw.
+type Raw struct {
```

Comment:
I don't think adding a `Raw` type here is a good idea, given that our BSON library already has a `Raw` type that is completely different than this. This `Raw` is actually more similar to our `RawValue`. Also, the docs for `Setter` below mention that `Raw.Unmarshal` can be used in a `SetBSON` implementation, but this type does not have a `Unmarshal` method.

I think a better choice would be to get rid of this type and have `SetBSON` take a `bson.RawValue`, which does have an `Unmarshal` method. Having this type doesn't actually prevent code changes for our users, as they still have to change their `SetBSON` implementations to take `mgocompat.Raw` instead of `bson.Raw`, but it does cause unnecessary confusion in my opinion.

Reply:
done